src/iter.rs

```diff
@@ -24,26 +22,19 @@ pub struct MiniconfIter<M: ?Sized, const L: usize, const TS: usize, S> {
     count: Option<usize>,
 }
 
-impl<M: ?Sized, const L: usize, const TS: usize, S> Default for MiniconfIter<M, L, TS, S> {
+impl<M, S, const L: usize, P> Default for MiniconfIter<M, S, L, P> {
     fn default() -> Self {
         Self {
             count: None,
-            miniconf: PhantomData,
-            spec: PhantomData,
+            marker: PhantomData,
             state: [0; L],
         }
     }
 }
 
-impl<M: ?Sized + Miniconf + SerDe<S>, const L: usize, const TS: usize, S>
-    MiniconfIter<M, L, TS, S>
-{
+impl<M: SerDe<S>, S, const L: usize, P> MiniconfIter<M, S, L, P> {
     pub fn metadata() -> Result<Metadata, IterError> {
         let meta = M::metadata(M::SEPARATOR.len_utf8());
-        if TS < meta.max_length {
```

Comment:
If we can't pre-emptively check the length, do we even want to precalculate the data length? I guess it's still potentially useful to the end user for initial setup.